### PR TITLE
pythonPackages.obspy: init at 1.2.2

### PR DESCRIPTION
**Pull request by matilde-ametrine**
###### Motivation for this change

ObsPy is not (yet) part of nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/obspy/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    decorator
+    future
+    lxml
+    matplotlib
+    numpy
+    requests
+    scipy
+    sqlalchemy
+  ];
+
```
Comment (risicle) [inline pkgs/development/python-modules/obspy/default.nix:34]: Could we have a `pythonImportsCheck` if we're not going to have the tests enabled?

Comment (matilde-ametrine) [reply pkgs/development/python-modules/obspy/default.nix:34]: Added `pythonImportsCheck = [ "obspy" ];`

File: pkgs/development/python-modules/obspy/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  # Tests require Internet access.
+  doCheck = false;
+
+  pythonImportsCheck = [ "obspy" ];
+
+  meta = with lib; {
+    description = "Python framework for seismological observatories";
+    homepage = https://www.obspy.org;
```
Comment (risicle) [inline pkgs/development/python-modules/obspy/default.nix:42]: I _think_ we're trying to deprecate the unquoted url format these days in favour of just plain strings.

**Comment by risicle** [comment]
Builds for me, binaries run, macos 10.15 & nixos x86_64.

File: pkgs/development/python-modules/obspy/default.nix (hunk 2 above)
Comment (matilde-ametrine) [reply pkgs/development/python-modules/obspy/default.nix]: Looks like you're right (https://github.com/NixOS/rfcs/pull/45). Changed.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -12602,5 +12602,15 @@
     github = "mbprtpmnr";
     githubId = 88109321;
   };
+  ametrine = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:12605]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (matilde-ametrine) [reply maintainers/maintainer-list.nix]: Done.
